docs/testing.md

```diff
@@ -0,0 +1,138 @@
+# Testing BMI
+
+Every Developer will need to test his/her feature or make sure that he/she
+did not break anything. So the test cases are situated in
+
+```
+/tests/
+```
+
+## Directory Structure
+
+This folder follows the directory structure of the project with additional
+folders like for example operation.py's test cases are situated in
+```
+/tests/einstein/operations/unit/...
+/tests/einstein/operations/integration/...
+```
+
+Every file must have a folder for itself which has additional folders called
+'unit' and 'integration'
+
+All the unit tests should be in unit folder and all integration tests should
+be in integration folder for that file
+
+All of these folders are python packages so they should have a \_\_init\_\_.py
+
+## Writing Tests
+
+Create a file in the correct location of the directory structure and make sure
+the name conforms to the regex 'test_\*.py'
+
+We are using python's unittest as our testing framework for now.
+
```

Comment:
just out of curiosity, is there a specific reason to use `unittest` instead of `pytest`. I don't have a strong opinion on this, it's just that I am familiar with `pytest`